### PR TITLE
Replace FormInput with AutoInput and refactored useFormFields to return an array of objects

### PR DESCRIPTION
**Pull request by MillanWangGadget**
- **UPDATES**
  - Removed FormInput components and replaced them with AutoInput
    - Polaris and MUI
  - Refactored useFormFields to return an array of objects instead of an array of arrays
    - `{ path:string, fieldMetadata:FieldMetadata }` feels less risky than using an array and trusting ourselves to pass in the proper index

## PR Checklist

- [ ] Important or complicated code is tested
- [ ] Any user facing changes are documented in the Gadget-side changelog
- [ ] Any immediate changes are slated for release in Gadget via a generated package dependency bump
- [ ] Versions within this monorepo are matching and there's a valid upgrade path

### CODE REVIEW

File: packages/react/src/auto/AutoForm.ts [hunk 1]
```diff
@@ -50,16 +50,27 @@ export const useValidationResolver = (metadata: ActionMetadata | undefined) => {
 export const useFormFields = (
   metadata: ActionMetadata | undefined | null,
   options: { include?: string[]; exclude?: string[]; fields?: string[]; select?: Record<string, any> }
-): (readonly [string, FieldMetadata])[] => {
+): readonly { path: string; fieldMetadata: FieldMetadata }[] => {
```
Comment (alanko0511) [inline packages/react/src/auto/AutoForm.ts:53]: `field.fieldMetadata` seems a bit weird. How about `field.metadata`?

**Review by alanko0511 — APPROVED** [review]

File: packages/react/src/auto/AutoForm.ts (hunk 1 above)
Comment (MillanWangGadget) [reply packages/react/src/auto/AutoForm.ts]: Updated to just be `metadata`

File: packages/react/src/auto/mui/inputs/MUIAutoInput.tsx [hunk 2]
```diff
@@ -19,36 +17,39 @@ const FieldTypeToInputType: Partial<Record<GadgetFieldType, TextFieldProps["type
...
+        <FormControlLabel label={fieldMetadata.name} control={props.children} />
       </FormGroup>
       {error && <FormHelperText>{error?.message}</FormHelperText>}
     </FormControl>
   );
 };
 
-export const MUIFormInput = (props: { path: string; field: FieldMetadata; control: Control<any> }) => {
+export const MUIAutoInput = (props: { field: string }) => {
+  const { path, metadata: fieldMetadata } = useFieldMetadata(props.field);
```
Comment (danroberts) [inline packages/react/src/auto/mui/inputs/MUIAutoInput.tsx:40]: Maybe we can just call this field metadata -- I think it's clear enough that it is the metadata for the field, and we can resolve the field/fieldMetadata ambiguity?

**Review by danroberts — APPROVED** [review]

File: packages/react/src/auto/mui/inputs/MUIAutoInput.tsx (hunk 2 above)
Comment (MillanWangGadget) [reply packages/react/src/auto/mui/inputs/MUIAutoInput.tsx]: Updated so that this is just `metadata` for all places that use this hook

**Pull request merged**